### DISCUSSION TITLE
Any ideas on how distributed Actor system could be implemented using Sento?

### DISCUSSION THREAD

**Discussion by svetlyak40wt** [mdbergmann/cl-gserver#90] [Ideas]
Hi @mdbergmann, do you have how to build a fault tolerant distributed actor system with Sento?

**Comment by mdbergmann** [comment]
There are two things necessary for this:

1. A concept that determines what should happen if an actor errors. In Erlang and Akka exists a concept called 'supervision'. This basically describes how 'worker' actors are created or what should happen if one of them errors. How the supervision works is usually configurable.

2. Remote support where an actor in the actor system can 'location transparently' send a message to another 'known' actor in another actor system. The actor systems can either be on the same host or on different hosts.

"2" doesn't exist (yet) in Sento, there is no remote support. I was thinking about this for a while and have some ideas how to implement it but didn't get around to do it. Would be interesting though.
As for "1", in Sento Actors can't error in a way that allows them to terminate. Any error/condition raised in the 'receive' is just simply handled and logged. But eventually it depends on the actor 'receive' implementation how errors are handled and so how fault tolerant the whole thing is.
